### PR TITLE
Use a queue to trigger checkin

### PR DESCRIPTION
**Pull request by swtools0**
This extra layer of indirection will allow us to, in the future, cleanly handle any errors thrown by handle-scheduled-checkin by simply creating a consumer of the DLQ.

### CODE REVIEW

File: .vscode/cspell.json [hunk 1]
```diff
@@ -16,9 +16,18 @@
     "openflights",
     "parens",
     "prettierrc",
+    "redrive",
     "slsw",
     "vite"
   ],
-  "ignoreWords": ["checkedin", "laxlga", "youre", "iatafilter", "mweb", "requestfinished"],
+  "ignoreWords": [
+    "checkedin",
```
Comment (aaron-pham) [inline .vscode/cspell.json:24]: what do these do?

File: src/handlers/handle-scheduled-checkin.ts [hunk 2]
```diff
@@ -1,40 +1,51 @@
...
+
+    // allow sending to dlq immediately
+    await adjustMessageVisibilityTimeout({
+      queueUrl: process.env.SCHEDULED_CHECKIN_READY_QUEUE_URL,
+      receiptHandle: event.Records[0].receiptHandle,
+      visibilityTimeout: 0
+    });
+
+    // send to dlq
     throw error;
```
Comment (aaron-pham) [inline src/handlers/handle-scheduled-checkin.ts:30]: What ends up going to the DLQ? The error itself?

File: src/scripts/tests/test-create-eventbridge-rule.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import console from 'console';
+import * as Luxon from 'luxon';
+import process from 'process';
+import { putRule, putTarget } from '../../lib/create-eventbridge-rule';
+import { generateCronExpressionUtc } from '../../lib/cron-utils';
+import { Reservation } from '../../lib/reservation';
+import * as Queue from '../../lib/scheduled-checkin-ready-queue';
+
+async function main() {
+  assert(process.argv.length === 3, 'Invalid parameters');
```
Comment (aaron-pham) [inline src/scripts/tests/test-create-eventbridge-rule.ts:13]: Does this assert happen after everything executes? Does this create resources in your AWS account that needs to be cleaned up eventually? 

Still new with AWS work and interested in how testing generally works in it!

File: .vscode/cspell.json (hunk 1 above)
Comment (swtools0) [reply .vscode/cspell.json:24]: These are words that are not really words but that we want to ignore with cspell. cspell will not consider them misspelled, but will also not make suggestions based on them.

File: src/handlers/handle-scheduled-checkin.ts (hunk 2 above)
Comment (swtools0) [reply src/handlers/handle-scheduled-checkin.ts:30]: The same message that this function received. So if we want to implement notifications, we should add the user's email address and/or password to the EventBridge rule's detail object, which will end up in the SQS message, which will be delivered to the DLQ consumer.

File: src/scripts/tests/test-create-eventbridge-rule.ts (hunk 3 above)
Comment (swtools0) [reply src/scripts/tests/test-create-eventbridge-rule.ts:13]: The assert is executed first.

Yes, this script will create an EventBridge rule and target. I suppose we could create an entire dev version of this stack for testing, but that doesn't seem worth it.

**Pull request merged**